OpenCL_C.txt

```diff
@@ -72,82 +72,141 @@ in OpenCL C.
+OpenCL C compilers that define the feature macro `+__opencl_c_pipes+` must
+also define the feature macro `+__opencl_c_generic_address_space+`.
 
 | `+__opencl_c_program_scope_global_variables+`
-| The OpenCL C compiler supports program scope variables in the global address space.
+| The OpenCL C compiler supports program scope variables in the global address
+space.
 
 | `+__opencl_c_read_write_images+`
-| The OpenCL C compiler supports reading from and writing to the same image object in a kernel.
+| The OpenCL C compiler supports reading from and writing to the same image
+object in a kernel.
+
+OpenCL C compilers that define the feature macro
+`+__opencl_c_read_write_images+` must also define the feature macro
+`+__opencl_c_images+`.
 
-OpenCL C compilers that define the feature macro `+__opencl_c_read_write_images+` must also define the feature macro `+__opencl_c_images+`.
 | `+__opencl_c_subgroups+`
-| The OpenCL C compiler supports built-in functions operating on sub-groupings of work-items.
+| The OpenCL C compiler supports built-in functions operating on sub-groupings
+of work-items.
 
 | `+__opencl_c_work_group_collective_functions+`
-| The OpenCL C compiler supports built-in functions that perform collective operations across a work-group.
+| The OpenCL C compiler supports built-in functions that perform collective
+operations across a work-group.
 
 |====
 
-In OpenCL C 3.0, feature macros must expand to the value `1` if the feature macro is defined by the OpenCL C compiler.
-A feature macro must not be defined if the feature is not supported by the OpenCL C compiler.
-A feature macro may expand to a different value in the future, but if this occurs the value of the feature macro must compare greater than the prior value of the feature macro.
+In OpenCL C 3.0, feature macros must expand to the value `1` if the feature
+macro is defined by the OpenCL C compiler. A feature macro must not be
+defined if the feature is not supported by the OpenCL C compiler. A feature
+macro may expand to a different value in the future, but if this occurs the
+value of the feature macro must compare greater than the prior value of the
+feature macro.
+
+[[extensions]]
+=== Extensions
+
+Optional functionality that is not defined in this document is referred to
+as extensions. Extensions are described in
+<<opencl-extension-spec,the OpenCL Extension Specification>>.
 
 [NOTE]
 --
-The OpenCL 3.0 C programming language described in this specification is provisional.
-It has been Ratified under the Khronos Intellectual Property Framework and is being made publicly available to enable review and feedback from the community.
-
-If you have feedback please create an issue on https://github.com/KhronosGroup/OpenCL-Docs/
+Prior to OpenCL 3.0 some optional features described in this document were
+referred to as optional core language extensions. Their presence could be
+indicated by the predefined extension macros. If any of the features has been
```

Comment:
> indicated by the predefined extension macros.

Is this intended to cover features such as `__IMAGE_SUPPORT__`?  Because that was never an extension.

Reply:
Ok, I didn't mean to include originally it but we can do if you think it is needed. The reason I didn't include it is mainly because it is documented in Predefined macros section. So technically both `__IMAGE_SUPPORT__` and `__opencl_c_images` are documented in OpenCL C explicitly.

Reply:
Yes, you're right, the text is fine as-is.